src/devices/rpLidar2/sdk_original/sdk/src/rplidar_driver.cpp

```diff
@@ -0,0 +1,1026 @@
+            return ans;
+        }
+
+        // verify whether we got a correct header
+        if (response_header.type != RPLIDAR_ANS_TYPE_MEASUREMENT) {
+            return RESULT_INVALID_DATA;
+        }
+
+        _u32 header_size = (response_header.size_q30_subtype & RPLIDAR_ANS_HEADER_SIZE_MASK);
+        if (header_size < sizeof(rplidar_response_measurement_node_t)) {
+            return RESULT_INVALID_DATA;
+        }
+
+        _isScanning = true;
+        _cachethread = CLASS_THREAD(RPlidarDriverSerialImpl, _cacheScanData);
+        if (_cachethread.getHandle() == 0) {
+            return RESULT_OPERATION_FAIL;
+        }
+    }
+    return RESULT_OK;
+}
+
+u_result RPlidarDriverSerialImpl::checkExpressScanSupported(bool & support, _u32 timeout)
+{
+    rplidar_response_device_info_t devinfo;
+
+    support = false;
+    u_result ans = getDeviceInfo(devinfo, timeout);
+
+    if (IS_FAIL(ans)) return ans;
+
+    if (devinfo.firmware_version >= ((0x1<<8) | 17)) {
+        support = true;
+        rplidar_response_sample_rate_t sample_rate;
+        getSampleDuration_uS(sample_rate);
+        _cached_sampleduration_express = sample_rate.express_sample_duration_us;
+        _cached_sampleduration_std = sample_rate.std_sample_duration_us;
+    }
+
+    return RESULT_OK;
+}
+
+u_result RPlidarDriverSerialImpl::startScanExpress(bool fixedAngle, _u32 timeout)
+{
+    u_result ans;
+    if (!isConnected()) return RESULT_OPERATION_FAIL;
+    if (_isScanning) return RESULT_ALREADY_DONE;
+
+    stop(); //force the previous operation to stop
+
+    {
+        rp::hal::AutoLocker l(_lock);
+
+        rplidar_payload_express_scan_t scanReq;
+        scanReq.working_mode = (fixedAngle?RPLIDAR_EXPRESS_SCAN_MODE_FIXANGLE:RPLIDAR_EXPRESS_SCAN_MODE_NORMAL);
+        scanReq.reserved = 0;
+
+        if (IS_FAIL(ans = _sendCommand(RPLIDAR_CMD_EXPRESS_SCAN,&scanReq, sizeof(scanReq)))) {
+            return ans;
+        }
+
+        // waiting for confirmation
+        rplidar_ans_header_t response_header;
+        if (IS_FAIL(ans = _waitResponseHeader(&response_header, timeout))) {
+            return ans;
+        }
+
+        // verify whether we got a correct header
+        if (response_header.type != RPLIDAR_ANS_TYPE_MEASUREMENT_CAPSULED) {
+            return RESULT_INVALID_DATA;
+        }
+
+        _u32 header_size = (response_header.size_q30_subtype & RPLIDAR_ANS_HEADER_SIZE_MASK);
+        if (header_size < sizeof(rplidar_response_capsule_measurement_nodes_t)) {
+            return RESULT_INVALID_DATA;
+        }
+
+        _isScanning = true;
+        _cachethread = CLASS_THREAD(RPlidarDriverSerialImpl, _cacheCapsuledScanData);
+        if (_cachethread.getHandle() == 0) {
+            return RESULT_OPERATION_FAIL;
+        }
+    }
+    return RESULT_OK;
+}
+
+
+u_result RPlidarDriverSerialImpl::startScan(bool force, bool autoExpressMode)
+{
+    bool isExpressModeSupported;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'isExpressModeSupported' can be reduced.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5162834)